python/torch_mlir/extras/fx_decomp_util.py

```diff
@@ -49,6 +49,7 @@
     torch.ops.aten.nan_to_num.default,
     torch.ops.aten.unbind,
     torch.ops.aten._scaled_dot_product_flash_attention_for_cpu,
+    torch.ops.aten.diag,
```

Comment:
I was confused by why this worked because there is no decomposition for this op in [torch/_decomp/decompositions.py](https://github.com/pytorch/pytorch/blob/main/torch/_decomp/decompositions.py). Turns out that PyTorch has _a lot_ of decompositions in [another file](https://github.com/pytorch/pytorch/blob/main/torch/_refs/__init__.py). Pretty nice.

Also, FWIW, I just realized that Torch-MLIR [does support `torch.ops.aten.diag_embed`](https://github.com/llvm/torch-mlir/pull/2885), which behaves exactly like `torch.diag` for 1D inputs.